llvm/lib/IR/AutoUpgrade.cpp

```diff
@@ -4262,16 +4267,29 @@ void llvm::UpgradeIntrinsicCall(CallBase *CI, Function *NewFn) {
       Rep = upgradeARMIntrinsicCall(Name, CI, F, Builder);
     } else if (IsAMDGCN) {
       Rep = upgradeAMDGCNIntrinsicCall(Name, CI, F, Builder);
-    } else if (IsDbg && CI->getModule()->IsNewDbgInfoFormat) {
-      upgradeDbgIntrinsicToDbgRecord(Name, CI);
+    } else if (IsDbg) {
+      // We might have decided we don't want the new format after all since
+      // requesting the upgrade; skip the conversion if that is the case, and
+      // check here to see if the intrinsic needs to be upgraded normally.
+      if (!CI->getModule()->IsNewDbgInfoFormat) {
```

Comment:
nit: I think this could be slightly more clear, e.g. "We might have decided we don't want the new format after all since requesting the upgrade" -> "We might have decided we don't want the new format after all between requesting the upgrade initially and now;"

(I think "since" is just a slightly ambiguous word used here)